### DISCUSSION TITLE
Data undefined after background duration

### DISCUSSION THREAD

**Discussion by haswalt** [urql-graphql/urql#2363] [Q&A] [answered]
Using ReactNative, a simple query returns data correctly first request. After leaving the app in the background for a period of time (~30 minutes) the requests no longer return data. This isn't dependent on caching.

This can be observed with cache disabled and a simple list of exchanges (dedup, error, multipartFetch). When enabling caching (document cache), it is observed as initial render fetching false, data defined. Then a re-render is triggered. fetching false, data undefined.

Calling refetch for the query results in fetching true but data remains undefined. Data is being returned by the server when testing with Insomnia.

Anyone else seeing this? Initially we saw this issue and did find issues related to the retry exchange and timers, this has been resolved by @kitten however I am not using retry exchange as we removed it until the issue was fixed.

As suggested when investigating the issue i've tried using unfetch polyfill but that doesn't seem to help.

urql: 2.2.0
@urql/core: 2.4.3
graphql: 16.3.0

**Comment by haswalt** [comment, marked as answer]
On further investigation I can replicate this issue with a simple component that is hidden and shown in a tab bar type layout. Once the component has been idle for a period of time (maybe an hour) then the request never gets data. fetching is fale but data is undefined.

Network polidy is cache-and-network

**Comment by kirtirajsinh** [comment]
I'm facing a similar kind of issue. On querying the data it returns the result the first time but on reloading the page it gives an undefined data with a `combined error` result. I'm querying data in a Next.js project. How Do I resolve it?
